### DISCUSSION TITLE
How to load images correctly ?

### DISCUSSION THREAD

**Discussion by antoinegreuzard** [wordplate/wordplate#343] [Q&A] [answered]
Hello,

How to load images on php file and scss file ?

I used the function in a previous question, but it's not very ok :
```
@function theme-url($path, $theme: 'wordplate') {
  @return url('/themes/#{$theme}/assets/#{$path}');
}
```

How to load images with resources folder ?
When I load images from scss : `background-image: url("/resources/static/images/image.jpg");` in scss, it's gave me this in front : 
background-image: url(/image-d18ff1d8.jpg);
The real path is `/public/themes/wordplate/assets/image-d18ff1d8.jpg` or `public/themes/wordplate/assets/images/image.jpg`

I use `<?= get_theme_file_uri( '/assets/images/image.jpg' ) ?>` in php.

Finally,  it's not working in npm run dev, here the result for the scss : `background-image: url(/resources/static/images/image.jpg);`
The assets folder not change on save, so I can't load image in php without npm run build every 2 seconds.

Thanks.

**Comment by vinkla** [comment]
Not sure. I try to put as many images in the HTML as possible. This might help: https://laravel.com/docs/9.x/vite#url-processing

**Reply by antoinegreuzard** [reply]
I did not succeed, can you give me an example? With hierarchy and image call in php file?

**Reply by vinkla** [reply]
No, I can't help debug your project.

**Comment by Kingo11** [comment]
I facing the same problem, my solution is to put my images directly in `/public/themes/wordplate/assets/my-image.jpg`
I'm using the `/resources` path only for css and js.

**Reply by antoinegreuzard** [reply]
Yes me too, my solution without using react, blade, vue, etc :
`get_theme_file_uri('/images/image.jpg') in php`

```
@function theme-url($path, $theme: 'wordplate'){
@return url('/themes/#{$theme}/#{$path}');
} in scss
```

I put images in my theme folder, in images/

**Reply by vinkla** [reply]
Anything you put in the `resources/static` directory should be copied over to the `theme/theme-name/assets` directory.

https://github.com/vinkla/wordplate/blob/3b2d76c13e8f4c14b6f1672507e742af3b9ef3aa/vite.config.js#L6

**Reply by Kingo11** [reply]
Yes anything put in `resources/static` is copied in the good folder and the build is working.
But when using the dev server, we cannot use the resource, it's embarassing.
Impossible to access, i was searching many solutions but nothing working.
My only solution is not using `resources/static`..
I continue searching, i think that's possible.

**Reply by vinkla** [reply]
Yeah, that is an issue I've have experienced as well. The static files are only copied when you run `npm run build`.

**Reply by antoinegreuzard** [reply]
It's not about laravel-vite-plugin ?

**Reply by vinkla** [reply]
Don't know. Haven't used `laravel-vite-plugin` with WordPlate.

**Reply by Kingo11** [reply]
Ok guys, i finally got it !
Here is my vite.config.js:
[https://github.com/Kingo11/viteJSconfig-wordpress/blob/master/vite.config.js](https://github.com/Kingo11/viteJSconfig-wordpress/blob/master/vite.config.js)

I added some commentaries to better understanding.
With that, images loaded successfully with dev and build.
Css example:
`html {
    background-image: url('../static/my-image.png');
}`
For html and php files just create a directive for images href with `http://localhost:5173/resources/static/` for dev and `http://localhost/themes/wordplate/assets/img/` for build.
I hope that will help you !

**Reply by antoinegreuzard** [reply]
Thanks, it's work for the css, but i don't understand for the php file in the theme.
We need to changes the href in dev to build mode ?

**Reply by Kingo11** [reply]
Ok nice.
For php juste use a directive like the function.php file to know if you are in local or prod:
`<?php
if (
    wp_get_environment_type() === 'local' &&
    is_array(wp_remote_get('http://localhost:5173/')) // is Vite.js running
) {
    $hrefImages = "http://localhost:5173/resources/static/";
} else {
    $hrefImages = get_theme_file_uri('assets/');
}?>
<img src="<?= $hrefImages."my-image.png" ?>" alt="">`

**Reply by antoinegreuzard** [reply]
thanks !

**Comment by antoinegreuzard** [comment, marked as answer]
**Solution** (thanks to @Kingo11 and @vinkla) :

https://github.com/antoinegreuzard/wordplate-vite.config.js
